recipe/meta.yaml

```diff
@@ -34,6 +34,8 @@ requirements:
     - setuptools
     - toml
     - typing-extensions
+    - poetry
+    - ruamel.yaml
```

Comment:
@mariusvniekerk what do you think about adding these two deps here? My guess is that they are both optional, right?

Reply:
ruamel got added since conda-lock now vendors conda :(

Reply:
We should probably bound poetry to `>=1.0,<1.2`

Reply:
> Poetry is optional but I believe we should ship with it for pip support. If we import pkg_resources we do need to add setuptools explicitly. Not sure about ruamel.yaml though.

We do import `pkg_resources`.  Wonder if we should switch over to 'importlib_metadata`

Reply:
gotcha, gonna constrain poetry then